### PR TITLE
Take a copy when making arrays for asdict

### LINKED ISSUE #1025: LWT dict has bad data if LWT object is deleted.
This works:
```python
>>> lwt = _tskit.LightweightTableCollection()
>>> tskit.TableCollection.fromdict(lwt.asdict())
<tskit.tables.TableCollection object at 0x7f116f75fb50>
```
How ever this doesn't:
```python
>>> lwt_dict = _tskit.LightweightTableCollection().asdict()
>>> tskit.TableCollection.fromdict(lwt_dict)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 2353, in fromdict
    tables.individuals.set_columns(**tables_dict["individuals"])
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 558, in set_columns
    self.ll_table.set_columns(
ValueError: Bad offset column encoding
```
this is basically the same as:
```python
>>> lwt = _tskit.LightweightTableCollection()
>>> lwt_dict = lwt.asdict()
>>> del lwt
>>> tskit.TableCollection.fromdict(lwt_dict)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 2353, in fromdict
    tables.individuals.set_columns(**tables_dict["individuals"])
  File "/home/benj/projects/tskit/python/tskit/tables.py", line 558, in set_columns
    self.ll_table.set_columns(
ValueError: Bad offset column encoding
```

I think this is because the dict encoding uses `PyArray_SimpleNewFromData` whose documentation says "Create an array wrapper around data pointed to by the given pointer. " Which to me sounds like no copy is taken and it is up to us to make sure the data has the same lifetime as any array object we've made. I guess the right thing to do here is to make a copy - although I find this unsatisfying and do wonder if we should grasp the nettle of dealing with this and `.tables` without a copy. I guess it only matters for very large tree sequences though, unless I've misunderstood?

In summary - everything has worked thus far as the lifetime of our LWTs has always been longer than the dicts they produce.

### PR DESCRIPTION
**Pull request by benjeffery**
## Description

The `tskit.LightweightTableCollection` and `tskit.TableCollection` were not taking copies of arrays, leading to lifecycle errors where the array could be deallocated.

Fixes #1025 

# PR Checklist:

- [x] Tests that fully cover new/changed functionality.
- [x] Documentation including tutorial content if appropriate.
- [x] Changelogs, if there are API changes.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/PR-1029/)

**Comment by benjeffery** [comment]
I've also modified the tests to use a fixture, this reduces the test time to 25% of what it was. This helps in the stress test which I have run to 2000 iterations without seeing any leaks.

File: python/lwt_interface/dict_encoding_testlib.py [hunk 1]
```diff
@@ -652,3 +597,32 @@ def test_top_level_metadata_schema(self):
...
+class TestLifecycle:
+    def test_unassigned_empty(self):
+        lwt_dict = lwt_module.LightweightTableCollection().asdict()
+        assert tskit.TableCollection.fromdict(lwt_dict) == tskit.TableCollection(-1)
+
+    def test_del_empty(self):
+        lwt = lwt_module.LightweightTableCollection()
+        lwt_dict = lwt.asdict()
+        del lwt
+        print(lwt_dict)
```
Comment (jeromekelleher) [inline python/lwt_interface/dict_encoding_testlib.py:611]: Stray print

**Review by jeromekelleher — APPROVED** [review]
LGTM!

File: python/lwt_interface/dict_encoding_testlib.py (hunk 1 above)
Comment (benjeffery) [reply python/lwt_interface/dict_encoding_testlib.py]: Dammit! So close!

**Pull request merged**